nodebuilder/share/constructors.go

```diff
@@ -57,3 +65,29 @@ func ensureEmptyCARExists(ctx context.Context, store *eds.Store) error {
 	}
 	return err
 }
+
+func peerManager(
+	headerSub libhead.Subscriber[*header.ExtendedHeader],
+	shrexSub *shrexsub.PubSub,
+	discovery *disc.Discovery,
+	host host.Host,
+	connGater *conngater.BasicConnectionGater,
+) *peers.Manager {
+	// TODO: find better syncTimeout duration?
+	return peers.NewManager(headerSub, shrexSub, discovery, host, connGater, modp2p.BlockTime*5)
+}
+
+func fullGetter(
+	store *eds.Store,
+	shrexGetter *getters.ShrexGetter,
+	ipldGetter *getters.IPLDGetter,
+) share.Getter {
+	return getters.NewCascadeGetter(
+		[]share.Getter{
+			getters.NewStoreGetter(store),
+			getters.NewTeeGetter(shrexGetter, store),
+			getters.NewTeeGetter(ipldGetter, store),
+		},
+		modp2p.BlockTime,
+	)
```

Comment:
We can have only one instance of TeeGetter if we wrap the CascadeGetter not shrex/ipld as per issue in https://github.com/celestiaorg/celestia-node/issues/1492

Reply:
yes, this was my original approach, but someone (I forgot who) pointed out that it would mean a successful retrieval from StoreGetter would try re-storing the EDS through TeeGetter - which would err. I think having two tee getters is more correct here

Reply:
Ok, but it would be noop for TeeGetter as it already has it there, and it will likely be a cache hit. Not a big deal though to have two TeeGetters, so let's just keep it if think its more correct

Reply:
I'd prefer it be a noop instead - check if stored, and if not, store, instead of having 2 tee getters.

Reply:
If it's too much work for right now, let's punt it and put it in an issue or something

Reply:
don't want to make tons of changes to what we are already running in TG, even if they seem trivial

Reply:
#1755